tour_model_eval/viz_bin_cluster.py

```diff
@@ -15,108 +15,86 @@
 import branca.colormap as cm
 
 
-def bins_map(bins, trips):
+def bins_map(bins, bins_list, trips):
```

Comment:
why can't you just pass in the bins that are selected by the `bins_list`?
having an `if` and `else` inside the function is still messy and I am not convinced that they are needed.

e.g. http://stackoverflow.com/questions/19252301/ddg#33356141

Reply:
I can create a `new_bins_list` that contains the selected bins, and let `bins` = `new_bins_list` in order to reuse the loop for general bins (in order, not selected). But then when I run the loop, it relies on the index in `bins` to plot the map. For example, if I pass[1,7,8], then `new_bins_list = [bins[1],bins[7],bin[8]]`, but the indices for `bins[1],bins[7],bin[8]` would be `0,1,2`. Then the map would use the first, second, and third color to plot (which are designed for the original bin index 0,1,2). That is different from the colormap, I think others will be confused.
The reason I need to pass in the bins is that I use it to fixed the colormap. Because I am actually using  `StepColormap` to make each bin has its own color. The color map highly relies on the indices. From the example above, if the indices is 0,1,2, I need to make the indices for colormap from 0 to 3 in order to have 3 colors, but in the colormap it will just show 0 1 2 3, it will not show 1 7 8.
Though the code looks repeated, but I think they will messy if I don't separate them. What do you think?

Reply:
the answer is simple and one I had hinted before. Associate the bins and colors before subsetting them.
`zip(bins,color)` will give you `(bin, color)` tuples - 

```
>>> colors = ["white", "gray", "green", "yellow"]
>>> bins = ["b1", "b2", "b3", "b4"]
>>> list(zip(bins, colors))
[('b1', 'white'), ('b2', 'gray'), ('b3', 'green'), ('b4', 'yellow')]
>>> list(zip(bins, colors))[2:4]
[('b3', 'green'), ('b4', 'yellow')]
>>> list(zip(bins, colors))[1:3]
[('b2', 'gray'), ('b3', 'green')]
>>>
```

Now, you can do:

```
for bin, color in sel_bin_colors:
....
```

Reply:
But my color is not a list like `colors = ["white", "gray", "green", "yellow"]`, I don't know what colors in the colormap, it depends on the steps/labels

Reply:
This is an example of how zip can work. The bins are also not "b1", "b2", "b3"...

It is a list of some colors, right? You can only index it because it is a list. 

You can zip your color list (whatever it is) with the bin list. Did you try it?